app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -320,31 +323,52 @@ private fun ConversationScreen(
         coroutineScope = conversationScreenState.coroutineScope,
-        menuItems = conversationScreenState.selectedMessage?.let { message ->
-            EditMessageMenuItems(
-                message = message,
-                hideEditMessageMenu = conversationScreenState::hideEditContextMenu,
-                onCopyClick = conversationScreenState::copyMessage,
-                onDeleteClick = onDeleteMessage,
-                onReactionClick = onReactionClick,
-                onDetailsClick = onMessageDetailsClick,
-                onReplyClick = messageComposerInnerState::reply,
-                onEditClick = messageComposerInnerState::toEditMessage,
-                onShareAsset = {
-                    conversationScreenState.selectedMessage?.header?.messageId?.let {
-                        conversationMessagesViewModel.shareAsset(context, it)
-                        conversationScreenState.hideEditContextMenu()
-                    }
-                },
-                onDownloadAsset = conversationMessagesViewModel::downloadAssetExternally,
-                onOpenAsset = conversationMessagesViewModel::downloadAndOpenAsset,
-            )
-        } ?: emptyList()
+        menuItems = when (val menuType = conversationScreenState.bottomSheetMenuType) {
+            is ConversationScreenState.BottomSheetMenuType.Edit -> {
+                EditMessageMenuItems(
+                    message = menuType.selectedMessage,
+                    hideEditMessageMenu = conversationScreenState::hideContextMenu,
+                    onCopyClick = conversationScreenState::copyMessage,
+                    onDeleteClick = onDeleteMessage,
+                    onReactionClick = onReactionClick,
+                    onDetailsClick = onMessageDetailsClick,
+                    onReplyClick = messageComposerInnerState::reply,
+                    onEditClick = messageComposerInnerState::toEditMessage,
+                    onShareAsset = {
+                        menuType.selectedMessage.header.messageId.let {
+                            conversationMessagesViewModel.shareAsset(context, it)
+                            conversationScreenState.hideContextMenu()
+                        }
+                    },
+                    onDownloadAsset = conversationMessagesViewModel::downloadAssetExternally,
+                    onOpenAsset = conversationMessagesViewModel::downloadAndOpenAsset
+                    )
+            }
+
+            is ConversationScreenState.BottomSheetMenuType.SelfDeletion -> {
+                SelfDeletionMenuItems(
+                    hideEditMessageMenu = conversationScreenState::hideContextMenu,
+                    currentlySelected = messageComposerInnerState.getSelfDeletionTime(),
+                    onSelfDeletionDurationChanged = { messageComposerInnerState.specifySelfDeletionTime(it) }
+                )
+            }
+
+            ConversationScreenState.BottomSheetMenuType.None -> emptyList()
```

Comment:
I like the idea but I think it would be even more boiler code to maintain, just because of passing all this lambdas to two functions instead :S so i moved it to a local variable instead